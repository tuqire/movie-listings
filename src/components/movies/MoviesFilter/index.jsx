import React from 'react'

export default ({
  genres,
  onGenreChanged,
  selectedGenres,

  onRatingsChange,
  ratings,
  selectedRating }) =>
  <React.Fragment>
    <div className="mb-4">
      <strong className="d-block mb-2">Lowest rating:</strong>
      <select value={selectedRating} onChange={onRatingsChange}>
        {ratings.map(selectedRating =>
          <option value={selectedRating} key={selectedRating}>{selectedRating}</option>)}
      </select>
    </div>
    <strong className="d-block mb-2">Genres:</strong>
    <div className="row">
      {genres.map(genre => (
        <div key={genre.id} className="col-sm-6 col-4 col-lg-3">
          <label>
            <input
              className={`genre-checkbox-${genre.id} mr-2`}
              type="checkbox"
              value={genre.id}
              name={`genre-checkbox-${genre.id}`}
              checked={selectedGenres.indexOf(genre.id) > -1} onChange={onGenreChanged}
            />
            {genre.name}
          </label>
        </div>
      ))}
    </div>
  </React.Fragment>
