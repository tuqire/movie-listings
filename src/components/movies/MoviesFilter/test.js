import React from 'react'
import Enzyme, { mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import MoviesFilter from './index'

Enzyme.configure({ adapter: new Adapter() })

const mockProps = {
  genres: [
    { id: 250, name: 'genre foo' },
    { id: 251, name: 'genre bar' },
    { id: 240, name: 'genre foobar' }
  ],
  onGenreChanged: jest.fn(),
  selectedGenres: [],
  onRatingsChange: jest.fn(),
  ratings: [1, 2, 3, 4, 5],
  selectedRating: 1
}

describe('components/movies/MoviesFilter', () => {
  describe('#MoviesFilter', () => {
    describe('matches snapshots', () => {
      it('when nothing in store', () => {
        const component = mount(<MoviesFilter {...mockProps} genres={[]} />)
        expect(component).toMatchSnapshot()
      })

      it('when genres in store', () => {
        const component = mount(<MoviesFilter {...mockProps} />)
        expect(component).toMatchSnapshot()
      })

      it('when selected genres in store', () => {
        const component = mount(<MoviesFilter {...mockProps} selectedGenres={[240, 251]} />)
        expect(component).toMatchSnapshot()
      })
    })

    describe('props', () => {
      let component

      beforeEach(() => {
        component = mount(<MoviesFilter {...mockProps} />)
      })

      describe('#onGenreChanged', () => {
        it('called when genre selected / unselected', () => {
          expect(mockProps.onGenreChanged).not.toHaveBeenCalled()

          component.find('.genre-checkbox-250').simulate('change', { foo: true })

          expect(mockProps.onGenreChanged).toHaveBeenCalledTimes(1)
          expect(mockProps.onGenreChanged).toHaveBeenCalledWith(expect.objectContaining({ foo: true }))
        })
      })

      describe('#onRatingsChange', () => {
        it('called when genre selected / unselected', () => {
          expect(mockProps.onGenreChanged).not.toHaveBeenCalled()

          component.find('select').simulate('change', { foo: true })

          expect(mockProps.onRatingsChange).toHaveBeenCalledTimes(1)
          expect(mockProps.onRatingsChange).toHaveBeenCalledWith(expect.objectContaining({ foo: true }))
        })
      })
    })
  })
})
