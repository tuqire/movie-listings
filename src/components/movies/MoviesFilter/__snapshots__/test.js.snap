// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/movies/MoviesFilter #MoviesFilter matches snapshots when genres in store 1`] = `
<Component
  genres={
    Array [
      Object {
        "id": 250,
        "name": "genre foo",
      },
      Object {
        "id": 251,
        "name": "genre bar",
      },
      Object {
        "id": 240,
        "name": "genre foobar",
      },
    ]
  }
  onGenreChanged={[Function]}
  onRatingsChange={[Function]}
  ratings={
    Array [
      1,
      2,
      3,
      4,
      5,
    ]
  }
  selectedGenres={Array []}
  selectedRating={1}
>
  <div
    className="mb-4"
  >
    <strong
      className="d-block mb-2"
    >
      Lowest rating:
    </strong>
    <select
      onChange={[Function]}
      value={1}
    >
      <option
        key="1"
        value={1}
      >
        1
      </option>
      <option
        key="2"
        value={2}
      >
        2
      </option>
      <option
        key="3"
        value={3}
      >
        3
      </option>
      <option
        key="4"
        value={4}
      >
        4
      </option>
      <option
        key="5"
        value={5}
      >
        5
      </option>
    </select>
  </div>
  <strong
    className="d-block mb-2"
  >
    Genres:
  </strong>
  <div
    className="row"
  >
    <div
      className="col-sm-6 col-4 col-lg-3"
      key="250"
    >
      <label>
        <input
          checked={false}
          className="genre-checkbox-250 mr-2"
          name="genre-checkbox-250"
          onChange={[Function]}
          type="checkbox"
          value={250}
        />
        genre foo
      </label>
    </div>
    <div
      className="col-sm-6 col-4 col-lg-3"
      key="251"
    >
      <label>
        <input
          checked={false}
          className="genre-checkbox-251 mr-2"
          name="genre-checkbox-251"
          onChange={[Function]}
          type="checkbox"
          value={251}
        />
        genre bar
      </label>
    </div>
    <div
      className="col-sm-6 col-4 col-lg-3"
      key="240"
    >
      <label>
        <input
          checked={false}
          className="genre-checkbox-240 mr-2"
          name="genre-checkbox-240"
          onChange={[Function]}
          type="checkbox"
          value={240}
        />
        genre foobar
      </label>
    </div>
  </div>
</Component>
`;

exports[`components/movies/MoviesFilter #MoviesFilter matches snapshots when nothing in store 1`] = `
<Component
  genres={Array []}
  onGenreChanged={[Function]}
  onRatingsChange={[Function]}
  ratings={
    Array [
      1,
      2,
      3,
      4,
      5,
    ]
  }
  selectedGenres={Array []}
  selectedRating={1}
>
  <div
    className="mb-4"
  >
    <strong
      className="d-block mb-2"
    >
      Lowest rating:
    </strong>
    <select
      onChange={[Function]}
      value={1}
    >
      <option
        key="1"
        value={1}
      >
        1
      </option>
      <option
        key="2"
        value={2}
      >
        2
      </option>
      <option
        key="3"
        value={3}
      >
        3
      </option>
      <option
        key="4"
        value={4}
      >
        4
      </option>
      <option
        key="5"
        value={5}
      >
        5
      </option>
    </select>
  </div>
  <strong
    className="d-block mb-2"
  >
    Genres:
  </strong>
  <div
    className="row"
  />
</Component>
`;

exports[`components/movies/MoviesFilter #MoviesFilter matches snapshots when selected genres in store 1`] = `
<Component
  genres={
    Array [
      Object {
        "id": 250,
        "name": "genre foo",
      },
      Object {
        "id": 251,
        "name": "genre bar",
      },
      Object {
        "id": 240,
        "name": "genre foobar",
      },
    ]
  }
  onGenreChanged={[Function]}
  onRatingsChange={[Function]}
  ratings={
    Array [
      1,
      2,
      3,
      4,
      5,
    ]
  }
  selectedGenres={
    Array [
      240,
      251,
    ]
  }
  selectedRating={1}
>
  <div
    className="mb-4"
  >
    <strong
      className="d-block mb-2"
    >
      Lowest rating:
    </strong>
    <select
      onChange={[Function]}
      value={1}
    >
      <option
        key="1"
        value={1}
      >
        1
      </option>
      <option
        key="2"
        value={2}
      >
        2
      </option>
      <option
        key="3"
        value={3}
      >
        3
      </option>
      <option
        key="4"
        value={4}
      >
        4
      </option>
      <option
        key="5"
        value={5}
      >
        5
      </option>
    </select>
  </div>
  <strong
    className="d-block mb-2"
  >
    Genres:
  </strong>
  <div
    className="row"
  >
    <div
      className="col-sm-6 col-4 col-lg-3"
      key="250"
    >
      <label>
        <input
          checked={false}
          className="genre-checkbox-250 mr-2"
          name="genre-checkbox-250"
          onChange={[Function]}
          type="checkbox"
          value={250}
        />
        genre foo
      </label>
    </div>
    <div
      className="col-sm-6 col-4 col-lg-3"
      key="251"
    >
      <label>
        <input
          checked={true}
          className="genre-checkbox-251 mr-2"
          name="genre-checkbox-251"
          onChange={[Function]}
          type="checkbox"
          value={251}
        />
        genre bar
      </label>
    </div>
    <div
      className="col-sm-6 col-4 col-lg-3"
      key="240"
    >
      <label>
        <input
          checked={true}
          className="genre-checkbox-240 mr-2"
          name="genre-checkbox-240"
          onChange={[Function]}
          type="checkbox"
          value={240}
        />
        genre foobar
      </label>
    </div>
  </div>
</Component>
`;
